### DISCUSSION TITLE
What is the best way to document how ISP circuits terminate?

### DISCUSSION THREAD

**Discussion by ghost** [netbox-community/netbox#10635] [Q&A]
I have a firewall that uses port 0 as the upstream to the ISP NID.
I have the ISP's NID as a device in my NetBox. 

When I terminate the ISP circuit on the firewall I connect a cable from port 0 to the ISP's circuit, but there doesn't seem to be a way for me to notate that the circuit is coming off of port 5 on the ISP NID.

I can connect the firewall port 0 interface to the NID port 5 interface, but then I'm just connecting 2 interfaces together with a cable, I'm not 'terminating' the ISP circuit.

What is the best way to notate this. When I do a cable trace I want to see both that the next hop from my firewalls port 0 interface is our ISP circuit with circuit ID xyz, but I also want to know that its connecting on the physical interface 5 on the ISP's NID so I don't have to manually trace a cable to figure out which port on the ISP's router our circuit is on.

**Comment by PieterL75** [comment]
You can create a Circuit for that.
Connect the Termination A to the port0 of the firewall and the TerminationZ to the port of the NID.
In that circuit you can document the provider, circuit details etc etc.

A cabletrace goes through a circui, showing the firewall, circuit and NID in one trace

https://docs.netbox.dev/en/stable/features/circuits/
https://docs.netbox.dev/en/stable/models/circuits/circuit/

https://demo.netbox.dev/circuits/circuits/30/

**Comment by ghost** [comment]
So is there any other way to do it?
I followed your instructions and end up with a cable trace that looks like this:

![image](https://user-images.githubusercontent.com/77940332/195634973-065dfa10-5137-463f-8fa7-c2abd5bb54ef.png)

The problem is, doing it this way makes me have (2) cables. 1 cable that connects from the firewall to the circuit, and a second cable to connect from the circuit to the NID. This is not reality. It is just 1 cable. Now in NetBox I have to have 2 cables and name them the same (but they will have different cable ID's).

**Reply by PieterL75** [reply]
Looks like you need another way to implement the circuit.
Create a regular cable between the firewall and the NID.
On the NID create an extra interface and link that to the circuit A side.
The Z side connects to a ' provider network ' and is marked as connected

**Comment by ghost** [comment]
Will I get the full trace of all of that when running a trace from the firewall? Or will I only get the circuit termination information if I run a cable trace on the NID?

**Reply by PieterL75** [reply]
Only when you run the trace from the NID
Unless, you create the ports on the NID as front and rear ports. Then you will trace from the firewall through the NID to the circuit

**Reply by ghost** [reply]
that's a great idea. I'll give that a try. thank you!

**Reply by ghost** [reply]
I guess last question. I'm having a brain fart right now. How do I link the front and rear port on the NID so it will trace through?

**Reply by PieterL75** [reply]
First create the rear port with one position
Then create the front port and link that to the rear port

**Reply by ghost** [reply]
Yes. My issue was you said front port but my head saw 'interface'.
Here is what it looks like now which matches reality.

![image](https://user-images.githubusercontent.com/77940332/195657155-49d78236-b04e-499b-95ba-370f5a689a7b.png)

**Reply by PieterL75** [reply]
Nice!

Glad we found a way to document what you expected

**Reply by goebelmeier** [reply]
Front and rear ports doesn't really fit the reality. Stick with interfaces and use the "bridge" functionality on interface level. Edit Gi0/0/0 set bridge mode to Gi0/0/1, edit Gi0/0/1 and set bridge mode to Gi0/0/0. that's basically what the ISP CPE does, it bridges traffic between two ports (and adding and removing S-VLANs and stuff like that, but that's irrelevant for the ISP customer).

And bridged interfaces show up in the cable trace, exactly like front and rear ports do, but they fit the reality more correct.

**Reply by PieterL75** [reply]
I never used the bridged interfaces, so that approach is also worth a try.
But as the NID is a ' transparent ' device (he does not control or configure the device), it is hard to tell what the reality is.

**Reply by ghost** [reply]
Bridge is exactly what I'm looking for. Now the NID doesn't have to have 'front' and 'rear' ports and I'm actually connecting interfaces together while still being able to trace through the NID to the circuit.

Thank you. I will use this method.

EDIT: one question. When I trace the path from the firewall towards the circuit it goes through the NID like I expect. But if I trace the cable from the circuit back the other way .. the firewall does not appear. It stops at the NID. Even though the interfaces are both bridged to each other. Is this expected behavior?

**Reply by bluikko** [reply]
Expected or not, IMO it should show the full path no matter where the trace is initiated from!
